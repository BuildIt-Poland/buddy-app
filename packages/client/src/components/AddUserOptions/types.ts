import { FabProps } from '@material-ui/core/Fab';

export interface AddUserOptionsProps extends FabProps {}
