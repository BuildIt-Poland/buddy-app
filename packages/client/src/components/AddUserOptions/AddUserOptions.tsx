import React from 'react';
import { useHistory } from 'react-router-dom';
import { UserRole } from '@buddy-app/schema';
import AddUserIcon from '@material-ui/icons/PersonAdd';
import PlusButton from 'atoms/PlusButton';
import DropDown, { ShowOptions } from 'components/DropDown';
import { ROUTES } from 'shared/routes';
import { AddUserOptionsProps } from './types';
import DICTIONARY from './dictionary';

const AddUserOptions = (props: AddUserOptionsProps) => {
  const history = useHistory();

  const renderOptions = () => [
    {
      text: DICTIONARY.OPTIONS.ADD_BUDDY,
      Icon: AddUserIcon,
      onClick: () => history.push(ROUTES.TALENT_ADD_BUDDY),
      access: {
        [UserRole.Talent]: true,
      },
    },
    {
      text: DICTIONARY.OPTIONS.ADD_TALENT,
      Icon: AddUserIcon,
      onClick: () => history.push(ROUTES.TALENT_ADD_TALENT),
      access: {
        [UserRole.Talent]: true,
      },
    },
  ];

  const renderAnchor = (showOptions: ShowOptions) => (
    <PlusButton {...props} onClick={showOptions} />
  );

  return (
    <DropDown
      id='add-user-options'
      renderOptions={renderOptions}
      renderAnchor={renderAnchor}
      marginThreshold={50}
    />
  );
};

export default AddUserOptions;
