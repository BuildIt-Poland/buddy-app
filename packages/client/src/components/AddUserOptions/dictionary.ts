interface Dictionary {
  OPTIONS: {
    ADD_BUDDY: string;
    ADD_TALENT: string;
  };
}

const DICTIONARY: Dictionary = {
  OPTIONS: {
    ADD_BUDDY: 'Add new Buddy',
    ADD_TALENT: 'Add new Talent',
  },
};

export default DICTIONARY;
