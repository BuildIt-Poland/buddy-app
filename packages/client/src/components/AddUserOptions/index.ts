import AddUserOptions from './AddUserOptions';

export default AddUserOptions;
