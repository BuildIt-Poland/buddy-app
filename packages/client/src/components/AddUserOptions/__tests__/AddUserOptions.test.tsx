import React from 'react';
import { create, act } from 'react-test-renderer';
import waitForExpect from 'wait-for-expect';
import { MemoryRouter, Route } from 'react-router';
import AddUserOptions from '../AddUserOptions';

jest.mock('@material-ui/icons/PersonAdd', () => 'AddUserIcon');
jest.mock('atoms/PlusButton', () => 'PlusButton');
jest.mock('components/DropDown', () => 'DropDown');

describe('Component - AddUserOptions', () => {
  const component = create(
    <MemoryRouter initialEntries={[]}>
      <Route path={'/buddy/newbies/:newbieId'}>
        <AddUserOptions />
      </Route>
    </MemoryRouter>
  );

  it('renders correctly', async () => {
    await act(async () => {
      await waitForExpect(() => {
        expect(component.toJSON()).toMatchSnapshot();
      });
    });
  });
});
