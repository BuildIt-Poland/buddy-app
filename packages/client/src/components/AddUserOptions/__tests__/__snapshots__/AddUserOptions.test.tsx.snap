// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - AddUserOptions renders correctly 1`] = `
<DropDown
  id="add-user-options"
  marginThreshold={50}
  renderAnchor={[Function]}
  renderOptions={[Function]}
/>
`;
